cmd/vclusterctl/cmd/platform/set/secret.go

```diff
@@ -0,0 +1,281 @@
+				},
+			},
+		}, metav1.CreateOptions{})
+		if err != nil {
+			return err
+		}
+
+		cmd.log.Donef("Successfully created secret %s with key %s", secretName, keyName)
+		return nil
+	}
+
+	// Update the secret
+	if secret.Spec.Data == nil {
+		secret.Spec.Data = map[string][]byte{}
+	}
+	secret.Spec.Data[keyName] = []byte(args[1])
+	_, err = managementClient.Loft().ManagementV1().ProjectSecrets(namespace).Update(ctx, secret, metav1.UpdateOptions{})
+	if err != nil {
+		return errors.Wrap(err, "update secret")
+	}
+
+	cmd.log.Donef("Successfully set secret key %s.%s", secretName, keyName)
+	return nil
+}
+
+func (cmd *SecretCmd) setSharedSecret(ctx context.Context, managementClient kube.Interface, args []string, namespace, secretName, keyName string) error {
+	secret, err := managementClient.Loft().ManagementV1().SharedSecrets(namespace).Get(ctx, secretName, metav1.GetOptions{})
+	if err != nil {
+		if !kerrors.IsNotFound(err) {
+			return errors.Wrap(err, "get secret")
+		}
+
+		secret = nil
+	}
+
+	if keyName == "" {
+		if secret == nil {
+			return fmt.Errorf(product.Replace("%w: for example 'loft set secret my-secret.key value'"), ErrNoSecret)
+		}
+		if len(secret.Spec.Data) == 0 {
+			return errors.Errorf(product.Replace("secret %s has no keys. Please specify a key like `loft set secret name.key value`"), secretName)
+		}
+
+		keyNames := []string{}
+		for k := range secret.Spec.Data {
+			keyNames = append(keyNames, k)
+		}
+
+		keyName, err = cmd.log.Question(&survey.QuestionOptions{
+			Question:     "Please select a secret key to set",
+			DefaultValue: keyNames[0],
+			Options:      keyNames,
+		})
+		if err != nil {
+			return errors.Wrap(err, "ask question")
+		}
+	}
+
+	// create the secret
+	if secret == nil {
+		_, err = managementClient.Loft().ManagementV1().SharedSecrets(namespace).Create(ctx, &managementv1.SharedSecret{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: secretName,
+			},
+			Spec: managementv1.SharedSecretSpec{
+				SharedSecretSpec: storagev1.SharedSecretSpec{
+					Data: map[string][]byte{
+						keyName: []byte(args[1]),
+					},
+				},
+			},
+		}, metav1.CreateOptions{})
+		if err != nil {
+			return err
+		}
+
+		cmd.log.Donef("Successfully created secret %s with key %s", secretName, keyName)
+		return nil
+	}
+
+	// Update the secret
+	if secret.Spec.Data == nil {
+		secret.Spec.Data = map[string][]byte{}
+	}
+	secret.Spec.Data[keyName] = []byte(args[1])
+	_, err = managementClient.Loft().ManagementV1().SharedSecrets(namespace).Update(ctx, secret, metav1.UpdateOptions{})
+	if err != nil {
+		return errors.Wrap(err, "update secret")
+	}
+
+	cmd.log.Donef("Successfully set secret key %s.%s", secretName, keyName)
+	return nil
+}
+
+func GetSharedSecretNamespace(namespace string) (string, error) {
+	if namespace == "" {
+		namespace = "loft"
+	}
+
+	return namespace, nil
+}
+
+func GetProjectSecretNamespace(project string) (string, error) {
+	return ProjectNamespacePrefix + project, nil
+}
```

Comment:
This function doesn't work with the latest platform version.
Project namespace prefixes must be fetched at runtime and included in the selves response.

Reply:
yeah good catch I missed a commit from loftctl